modules/core/keeper/msg_server.go

```diff
@@ -40,9 +41,33 @@ func (k *Keeper) CreateClient(ctx context.Context, msg *clienttypes.MsgCreateCli
 		return nil, err
 	}
 
+	// set the client creator so that eureka counterparty can be set by same relayer
+	k.ClientKeeper.SetClientCreator(ctx, clientID, sdk.AccAddress(msg.Signer))
+
 	return &clienttypes.MsgCreateClientResponse{ClientId: clientID}, nil
 }
 
+// RegisterCounterparty will register the eureka counterparty info for the given client id
+// it must be called by the same relayer that called CreateClient
+func (k *Keeper) RegisterCounterparty(ctx context.Context, msg *clienttypes.MsgRegisterCounterparty) (*clienttypes.MsgRegisterCounterpartyResponse, error) {
+	creator := k.ClientKeeper.GetClientCreator(ctx, msg.ClientId)
+	if !creator.Equals(sdk.AccAddress(msg.Signer)) {
```

Comment:
Should you instead have the creator become a string? Should be consistent with how we do these comparisons elsewhere?

Reply:
Left as is now, since it that case, we may as well just have the keeper return a string. Maybe we can do later as a small followup if really necessary